pkgs/development/compilers/go/1.21.nix

```diff
@@ -132,34 +131,24 @@ stdenv.mkDerivation rec {
     ulimit -a
 
     pushd src
-    ./make.bash
+    bash make.bash -no-banner -distpack
     popd
+
     runHook postBuild
   '';
 
-  preInstall = ''
-    # Contains the wrong perl shebang when cross compiling,
-    # since it is not used for anything we can deleted as well.
-    rm src/regexp/syntax/make_perl_groups.pl
-  '' + (if (stdenv.buildPlatform.system != stdenv.hostPlatform.system) then ''
-    mv bin/*_*/* bin
-    rmdir bin/*_*
-    ${lib.optionalString (!(GOHOSTARCH == GOARCH && GOOS == GOHOSTOS)) ''
-      rm -rf pkg/${GOHOSTOS}_${GOHOSTARCH} pkg/tool/${GOHOSTOS}_${GOHOSTARCH}
-    ''}
-  '' else lib.optionalString (stdenv.hostPlatform.system != stdenv.targetPlatform.system) ''
-    rm -rf bin/*_*
-    ${lib.optionalString (!(GOHOSTARCH == GOARCH && GOOS == GOHOSTOS)) ''
-      rm -rf pkg/${GOOS}_${GOARCH} pkg/tool/${GOOS}_${GOARCH}
-    ''}
-  '');
-
+  # NB the first line of the VERSION file is the Go version, additional lines
+  # are 'key value' pairs setting other data.
+  # See also https://pkg.go.dev/cmd/distpack
   installPhase = ''
     runHook preInstall
-    mkdir -p $GOROOT_FINAL
-    cp -a bin pkg src lib misc api doc go.env $GOROOT_FINAL
-    mkdir -p $out/bin
-    ln -s $GOROOT_FINAL/bin/* $out/bin
+
+    mkdir -p $out/{share,bin}
+    tar -C $out/share -x -z -f "pkg/distpack/$(sed 1q VERSION).$GOOS-$GOARCH.tar.gz"
+    ln -s $out/share/go/bin/* $out/bin
+
+    cp -a pkg/distpack $dist
```

Comment:
Why is `pkg/distpack` being packaged as well?

Reply:
Mostly with the intent to add passthru.tests to verify that (at least) source tarball builds are identical to official releases. That is, compare `$dist/go1.21.1.src.tar.gz` to `$src`. See also https://github.com/golang/go/issues/62053

>One of the goals of Go 1.21's reproducible builds effort was to have the binaries we post on go.dev/dl match the binaries that other distributions post after building from source. Of course, to make that the case, the other distributions need to not patch the source.

Although that is not possible right now because `patches` 😅

Now that I think about it, comparing source tarball does not really make sense, especially if we eventually remove patches (e.g. by implementing required features upstream). I’ll remove `dist` output, also because files are compressed and that doesn’t really play well with Nix dependency tracking.

Reply:
Removed `dist` output.